### PR TITLE
[ZEPPELIN-1316] Zeppelin can not start due to an incorrect Interpreter Setting	

### PR DESCRIPTION
**Pull request by cloverhearts**
### What is this PR for?
If there are problems with the information defined in the Interpreter Setting (conf / interpreter.json), Interpreter Group during the defined file does not exist, can not run Zeppelin.
Thus, it was an exception handling for that part.

In order to address this issue in the current version it was restarted after deleting Interpreter.json.

### What type of PR is it?
Bug Fix

### Todos
- [x] - check to Interpreter Setting is null

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1316

### How should this be tested?

1. Modify the following part of the Interpreter.json file.
```json
{
      "interpreterGroup": [
        {    
          "name": "spark",
          "class": "fake and invalid class path more more...", // invalid value
          "defaultInterpreter": false
        }
}
```

2. 
After restarting the Zeppelin, Zeppelin Confirm that normally works.
Version of the PR is to be normally started up.
It should also be display all possible Interpreter Web UI.

#### before
error

#### after
correct

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java [hunk 1]
```diff
@@ -365,9 +368,14 @@ private void loadFromFile() throws IOException {
       setting.getOption().setRemote(true);
 
       // Update transient information from InterpreterSettingRef
-      // TODO(jl): Check if reference of setting is null
-
-      setting.setPath(interpreterSettingsRef.get(setting.getGroup()).getPath());
+      interpreterSettingObject = interpreterSettingsRef.get(setting.getGroup());
+      if (interpreterSettingObject == null) {
+        logger.info("can't get InterpreterSetting " +
+          "Information From loaded Interrpeter Setting Ref - {} ", setting.getGroup());
```
Comment (AhyoungRyu) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java]: There is typo :)
`Interrpeter` -> `Interpreter`

**Comment by AhyoungRyu** [comment]
@cloverhearts Thanks for the patch! I faced this wired 503 error several times.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java (hunk 1 above)
Comment (cloverhearts) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java]: oh.. Thank you!
my mistake. :)

**Comment by bzz** [comment]
Thank you @cloverhearts ! That happened to me as well.

Looks great to me.

\cc @jongyoul for review

**Comment by jongyoul** [comment]
This will ignore non-exist setting if it doesn't exist anymore in `interpreter/*`. I think it looks realistic in production level because admin usually delete that interpreter setting if it doesn't exists. LGTM

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java [hunk 2]
```diff
@@ -365,9 +368,14 @@ private void loadFromFile() throws IOException {
       setting.getOption().setRemote(true);
 
       // Update transient information from InterpreterSettingRef
-      // TODO(jl): Check if reference of setting is null
-
-      setting.setPath(interpreterSettingsRef.get(setting.getGroup()).getPath());
+      interpreterSettingObject = interpreterSettingsRef.get(setting.getGroup());
+      if (interpreterSettingObject == null) {
+        logger.info("can't get InterpreterSetting " +
```
Comment (felixcheung) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java]: logger.warn?

Comment (cloverhearts) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java]: warning log?
Okay, i will modify it.

**Comment by bzz** [comment]
Thank you for prompt review!

Merging _to master_ if there is no further discussion

**Comment by bzz** [comment]
CI is green now, so merging it in

**Pull request closed without merging**